app/Models/Ldap.php

```diff
@@ -95,63 +95,35 @@ public static function findAndBindUserLdap($username, $password)
         $connection = self::connectToLdap();
         $ldap_username_field = $settings->ldap_username_field;
         $baseDn = $settings->ldap_basedn;
-        $userDn = $ldap_username_field.'='.$username.','.$settings->ldap_basedn;
-
-        if ($settings->is_ad == '1') {
-            // Check if they are using the userprincipalname for the username field.
-            // If they are, we can skip building the UPN to authenticate against AD
-            if ($ldap_username_field == 'userprincipalname') {
-                $userDn = $username;
-            } else {
-                // TODO - we no longer respect the "add AD Domain to username" checkbox, but it still exists in settings.
-                // We should probably just eliminate that checkbox to avoid confusion.
-                // We let it sit in the DB, unused, to facilitate people downgrading (if they decide to).
-                // Hopefully, in a later release, we can remove it from the settings.
-                // This logic instead just means that if we're using UPN, we don't append ad_domain, if we aren't, then we do.
-                // Hopefully that should handle all of our use cases, but if not we can backport our old logic.
-                $userDn = ($settings->ad_domain != '') ? $username.'@'.$settings->ad_domain : $username.'@'.$settings->email_domain;
-            }
-        }
 
+        // FIND the user first, then attempt to log in as them using the appropriate attribute
         $filterQuery = $settings->ldap_auth_filter_query.$username;
```

Comment:
Instead of concatenating the `$username`, I think it would be better to provide a string template in the config and then inserting the username into the template, this would allow `ldap_auth_filter_query` like:

```
| (uid=$username) (mail=$username)
```
which would enable people to loin with either uid or email

Reply:
It would also enable using only a single config field for what is now "ldap_filter" and "ldap_auth_filter_query"